### PR TITLE
[libsodium] Fix cross compile build

### PR DESCRIPTION
**Pull request by sphawk**
**Describe the pull request**
Fix cross compile build (host:windows, guest:android / triplet:arm64-android)

### CODE REVIEW

File: ports/libsodium/CMakeLists.txt [hunk 1]
```diff
@@ -257,7 +257,7 @@ else ()
     sodium_check_func(getpid HAVE_GETPID)
 
     if (VCPKG_TARGET_ARCHITECTURE STREQUAL x64)
-        check_c_source_runs(
+        check_c_source_compiles(
```
Comment (JackBoosY) [inline ports/libsodium/CMakeLists.txt:260]: Can you provide more details about this?
`check_c_source_runs` also check whether the test excutable run.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/libsodium/CMakeLists.txt (hunk 1 above)
Comment (dg0yt) [reply ports/libsodium/CMakeLists.txt:260]: In general, `check_c_source_runs` is a problem for true cross builds. And this PR is named "Fix cross compile build".

Comment (sphawk) [reply ports/libsodium/CMakeLists.txt:260]: @dg0yt  you're right. I want to build arm64-android version of libsodium on windows x64. (using NDK)
But windows cannot run arm64-android binary and can't use check_c_source_runs.
All "check_c_source_runs" parts doesn't see execution result, but only see it is compiled normally.
And I changed "check_c_source_runs" to "check_c_source_compiles" and it works.

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the fix!

**Pull request merged**